client/network/src/protocol/sync.rs

```diff
@@ -230,6 +236,8 @@ pub struct ChainSync<B: BlockT> {
 	/// Enable importing existing blocks. This is used used after the state download to
 	/// catch up to the latest state while re-importing blocks.
 	import_existing: bool,
+	/// Gap download process.
+	gap_sync: Option<GapSync<B>>,
```

Comment:
I would prefer to keep them separate. I see warp sync and old block download as two independent features. Even though the only way you can currently end up with the block gap is the warp sync, this may change in the future. I.e. when importing the state from a file or chain spec snapshot. Or after disabling block body pruning.